### PR TITLE
LPS-132441 Remove MBStatsUser

### PR DESCRIPTION
**Pull request by ctang3**
Continued from https://github.com/Preston-Crary/liferay-portal/pull/1342

Took a bit longer than it should because I initially removed the entire MBStatsUser entity in service.xml by mistake and was puzzled why ServiceBuilder wasn't removing things

### CODE REVIEW

File: modules/apps/message-boards/message-boards-service/src/main/java/com/liferay/message/boards/service/impl/MBStatsUserLocalServiceImpl.java [hunk 1]
```diff
@@ -192,11 +205,12 @@ public long getMessageCountByUserId(long userId) throws PortalException {
...
 					MBMessageTable.INSTANCE.categoryId.neq(
 						MBCategoryConstants.DISCUSSION_CATEGORY_ID)
+				).and(
+					MBMessageTable.INSTANCE.status.eq(
+						WorkflowConstants.STATUS_APPROVED)
 				)
 			).groupBy(
 				MBMessageTable.INSTANCE.userId
-			).having(
-				countExpression.gt(0L)
```
Comment (ctang3) [inline modules/apps/message-boards/message-boards-service/src/main/java/com/liferay/message/boards/service/impl/MBStatsUserLocalServiceImpl.java]: Removed this `.having(...)` clause because if it was included in the query in `getStatsUser` and the query returned no matching results, the following gets thrown:
`{level=ERROR, loggerName=org.hibernate.util.JDBCExceptionReporter, message=user lacks privilege or object not found: MESSAGECOUNT `

I figure that problem would apply to this query in `getStatsUsersByGroupId` as well.

I think its because the returned list of results is empty and therefore technically doesn't have any count of `messageId` to alias as a `messageCount` column?

File: modules/apps/message-boards/message-boards-service/src/main/java/com/liferay/message/boards/service/impl/MBStatsUserLocalServiceImpl.java [hunk 2]
```diff
@@ -154,8 +154,21 @@ public long getMessageCountByUserId(long userId) throws PortalException {
 				).and(
 					MBMessageTable.INSTANCE.categoryId.neq(
 						MBCategoryConstants.DISCUSSION_CATEGORY_ID)
+				).and(
+					MBMessageTable.INSTANCE.status.eq(
+						WorkflowConstants.STATUS_APPROVED)
```
Comment (ctang3) [inline modules/apps/message-boards/message-boards-service/src/main/java/com/liferay/message/boards/service/impl/MBStatsUserLocalServiceImpl.java:159]: Added this clause here and in the `getStatsUsersByGroupId` query so drafts and such aren't counted in response to failing draft-related tests in `MBStatsUserLocalServiceTest`

File: modules/util/portal-tools-sample-sql-builder/src/main/java/com/liferay/portal/tools/sample/sql/builder/DataFactory.java [hunk 3]
```diff
@@ -4613,32 +4611,6 @@ public MBMessageModel newMBMessageModel(
 		return mbMessageModels;
 	}
 
-	public MBStatsUserModel newMBStatsUserModel(long groupId) {
```
Comment (Preston-Crary) [inline modules/util/portal-tools-sample-sql-builder/src/main/java/com/liferay/portal/tools/sample/sql/builder/DataFactory.java:4616]: Need to update default_user.ftl since it uses this method.

File: modules/apps/message-boards/message-boards-test/src/testIntegration/java/com/liferay/message/boards/service/test/MBStatsUserLocalServiceTest.java [hunk 4]
```diff
@@ -169,11 +169,15 @@ protected void addMessage(boolean approved) throws Exception {
 			approved, serviceContext);
 	}
 
-	protected int getStatsUserMessageCount() throws Exception {
+	protected long getStatsUserMessageCount() throws Exception {
 		Object[] statsUser = MBStatsUserLocalServiceUtil.getStatsUser(
```
Comment (Preston-Crary) [inline modules/apps/message-boards/message-boards-test/src/testIntegration/java/com/liferay/message/boards/service/test/MBStatsUserLocalServiceTest.java:173]: Change this to do Long.intValue() instead of changing the return type.

File: modules/apps/message-boards/message-boards-service/src/main/java/com/liferay/message/boards/service/impl/MBStatsUserLocalServiceImpl.java [hunk 5]
```diff
@@ -154,8 +154,21 @@ public long getMessageCountByUserId(long userId) throws PortalException {
...
 				)
+			).groupBy(
+				MBMessageTable.INSTANCE.userId
+			).orderBy(
+				countExpression.descending()
 			));
+
+		if (statsUser.isEmpty()) {
```
Comment (Preston-Crary) [inline modules/apps/message-boards/message-boards-service/src/main/java/com/liferay/message/boards/service/impl/MBStatsUserLocalServiceImpl.java:167]: Why is this part needed?

File: modules/apps/message-boards/message-boards-service/src/main/java/com/liferay/message/boards/service/impl/MBStatsUserLocalServiceImpl.java (hunk 1 above)
Comment (Preston-Crary) [reply modules/apps/message-boards/message-boards-service/src/main/java/com/liferay/message/boards/service/impl/MBStatsUserLocalServiceImpl.java]: Yeah it's not needed since we're only querying the MBMessage table, I think I had that at one point because I was doing useless join on User.  Are you testing with hypersonic? Just trying to figure out how I didn't see the error when I was testing.

Comment (ctang3) [reply modules/apps/message-boards/message-boards-service/src/main/java/com/liferay/message/boards/service/impl/MBStatsUserLocalServiceImpl.java]: I am using hypersonic (although I do want to switch to mysql the next time I rebuild, if I don't forget)

File: modules/apps/message-boards/message-boards-test/src/testIntegration/java/com/liferay/message/boards/service/test/MBStatsUserLocalServiceTest.java [hunk 6]
```diff
@@ -169,11 +169,15 @@ protected void addMessage(boolean approved) throws Exception {
...
 			_group.getGroupId(), TestPropsValues.getUserId());
 
-		return (Integer)statsUser[1];
+		if (statsUser == null) {
+			return 0;
+		}
+
+		return (long)statsUser[1];
```
Comment (ctang3) [inline modules/apps/message-boards/message-boards-test/src/testIntegration/java/com/liferay/message/boards/service/test/MBStatsUserLocalServiceTest.java:180]: Return 0 if statsUser is null.

File: modules/apps/message-boards/message-boards-service/src/main/java/com/liferay/message/boards/service/impl/MBStatsUserLocalServiceImpl.java (hunk 5 above)
Comment (ctang3) [reply modules/apps/message-boards/message-boards-service/src/main/java/com/liferay/message/boards/service/impl/MBStatsUserLocalServiceImpl.java:167]: I added it for the case where no results are found by the the DSLQueary since it returns a List<Object[]> even if empty (and then in `MBStatsUserLocalServiceTest.getStatsUserMessageCount()` I check for null and return 0 if so): https://github.com/Preston-Crary/liferay-portal/pull/1350/files/3f400c24582874f1f6c295e08389c67c1e19bce3..afbdcad72e15930ee59b57debfe5685aac03b37e#r639516575

Maybe there's a better way to do it?

**Comment by ctang3** [comment]
Please see https://github.com/Preston-Crary/liferay-portal/pull/1354
